forbidigo/forbidigo.go

```diff
@@ -103,7 +103,12 @@ type visitor struct {
 	issues    []Issue
 }
 
-func (l *Linter) Run(fset *token.FileSet, typesInfo *types.Info, nodes ...ast.Node) ([]Issue, error) {
+// Deprecated: Use RunWithTypes
+func (l *Linter) Run(fset *token.FileSet, nodes ...ast.Node) ([]Issue, error) {
+	return l.RunWithTypes(fset, nil, nodes...)
+}
+
+func (l *Linter) RunWithTypes(fset *token.FileSet, typesInfo *types.Info, nodes ...ast.Node) ([]Issue, error) {
```

Comment:
https://github.com/ashanbrown/forbidigo/pull/22 then will add `pkg *types.Package` - do we then change the name again or interpret "WithTypes" as "with information from the types package"? I suspect you meant "with types info" here, in which case the function would have to become "RunWithTypesAndPkg" or something.

How about putting the args into a struct:
```
type RunConfig struct {
    fset *token.FileSet // required
    typesInfo *types.Info // optional, "pkg" rules won't work without it
    pkg *types.Package // optional, "file" rules won't work without it
}

func (l *Linter) RunWithConfig(config RunConfig, nodes ...ast.Node) ([]Issue, error) {
...
```

Reply:
The advantage of the struct is that new optional fields can be added for future extensions without an API break.

Reply:
Done.